enhancements/monitoring/monitoring-stack-operator.md

```diff
@@ -0,0 +1,490 @@
+
+## Motivation
+
+Red Hat managed services are deployed in OSD clusters and run alongside user workloads.
+
+Even though there are two monitoring stacks available in OSD clusters, neither is suitable for managed services. In OSD, Platform Monitoring is exclusively owned and managed by Platform SRE. It is customized to their needs and no modifications are allowed.
+Because of this, managed tenant owners (MTOs) cannot create additional ServiceMonitors, Prometheus rules and other monitoring resources on metrics in the Platform Monitoring domain.
+
+User Workload Monitoring was designed to be primarily used by customers for monitoring their own workloads. Managed services are discouraged from using this stack since they could easily get DOS-ed by high cardinality data from end-users.
+Similarly, the configuration of the Prometheus instance itself and Alerting and Routing, would be shared with customers and other managed service owners.
+
+As a result, each MTOs currently needs to deploy its own self-developed monitoring solutions alongside its service.
+Current options like [Prometheus Operator](https://operatorhub.io/operator/prometheus) (PO) through OLM provide inital value, but they require expertise on how to build a full-fledged monitoring stack with multi-tenancy and high availability in mind (RBAC proxies, HA, Thanos Querier, Ruler etc.).
+This imposes an extra cost on teams developing and maintaining a managed service and increases time-to-market for new services.
+
+### Goals
+
+This enhancement aims to relieve the burden from managed service owners by providing a ready-made monitoring solution. Such a solution would encapsulate existing patterns already present in Platform Monitoring and User Workload monitoring.
+Furthermore, it will reduce the operational complexity of running a managed service since the Managed Tenant SREs will not need to familiarise themselves with a new monitoring stack for each managed service.
+In future it can be used to allow customers to define their monitoring stacks. UWM could use it in future underneath too. In details:
+
+* Provide a way for managed services to scrape metrics from their own services and from platform components into prometheus-compatible storage.
+* Allow managed service owners to query metrics from their services and platform components from prometheus-compatible storage.
+* Provide a way for managed services to define alerting and recording rules on their own metrics and a subset of platform metrics.
+* Provide a way for managed services to create dashboards for monitoring their services.
+* Allow aggregating portions of managed service metrics at a multi-cluster level, e.g. for reporting SLOs.
+* Allow longer retention for portions of data (e.g. calculating 1 month burn down SLO budget).
+* Allow analytics on portion of managed service data in conjuction with Telemeter data
+* End customers using UWM should not see managed service metrics.
```

Comment:
Is this the same as "end customers using a managed service on their cluster that their workload talks to can't see the metrics of the managed service control plane?"  ultimately we want to ensure higher levels can aggregate workload metrics with the service metrics they consume (which may accidentally or deliberately be running on the same clusters as the workloads) - so just checking what this constraint means (i'm assuming the most minimal limit - "accidentally see")

Reply:
It is only about UWM, so user workload monitoring is assumed as things that can be used by customers and we don't want them to get confused by addons data.

Reply:
Does this imply that managed services that want to expose users to metrics will need to do so through some other mechanism?

Reply:
I think it's orthogonal to what we discuss here with @smarterclayton  but good question too.

We answered it in `Non Goals`:

> * Expose managed service metrics to end-users. We know e.g. OCS addon wants this, but this can be achieved on top of the proposed solution using client or backend side filtering for data isolation if needed (e.g. prom-label-proxy). The monitoring team can do this in the next iteration.